File: php/src/Advertisements/CivicCenter/Domain/ValueObjects/CivicCenterName.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisements\CivicCenter\Domain\ValueObjects;

use Demo\App\Advertisements\Advertisement\Domain\Exceptions\DescriptionEmptyException;
use Demo\App\Advertisements\Advertisement\Domain\Exceptions\DescriptionTooLongException;

final readonly class CivicCenterName
{
    /**
     * @throws DescriptionEmptyException
     * @throws DescriptionTooLongException
     */
    public function __construct(private string $value)
    {
        $this->validate($value);
    }

    public function value(): string
    {
        return $this->value;
    }

    /**
     * @throws DescriptionTooLongException
     * @throws DescriptionEmptyException
     */
    private function validate(string $value): void
    {
        if (mb_strlen($value) === 0) {
            throw DescriptionEmptyException::build();
        }

        if (mb_strlen($value) > 200) {
            throw DescriptionTooLongException::withLongitudeMessage($this->value);
        }
    }
}

File: kotlin/src/main/kotlin/advertisements/civiccenter/domain/value_object/CivicCenterName.kt
package advertisements.civiccenter.domain.value_object

import advertisements.advertisement.domain.exceptions.DescriptionEmptyException
import advertisements.advertisement.domain.exceptions.DescriptionTooLongException

class CivicCenterName(private val value: String) {

    init {
        validate(value)
    }

    fun value(): String {
        return value
    }

    @Throws(DescriptionEmptyException::class, DescriptionTooLongException::class)
    private fun validate(value: String) {
        if (value.isEmpty()) {
            throw DescriptionEmptyException.build()
        }

        if (value.length > 200) {
            throw DescriptionTooLongException.withLongitudeMessage(value)
        }
    }
}
